### PR TITLE
GET and POST collection API

### PR DESCRIPTION
**Pull request by LJT-512**
- Add GET collection API to allow users to see their collection based on the user id with pagination
- Add POST collection API to allow users to add and remove collection for a specific product id
- Error handling
   - cannot add collection for non-existing products
   - a user cannot add collection for a product id more than once

### CODE REVIEW

**Comment by LJT-512** [comment]
@marvelshan 再請你幫我看看這個 PR 感謝

File: server/swagger-output.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
     "description": "Stylish co-working project",
     "version": "1.0.0"
   },
-  "host": "35.72.177.254:3000",
+  "host": "undefined:3000",
```
Comment (nauish) [inline server/swagger-output.json]: 這個地方會不會掛掉XD

**Comment by nauish** [comment]
辛苦了！結果collection API要回的資料最多QQ

File: server/swagger-output.json (hunk 1 above)
Comment (LJT-512) [reply server/swagger-output.json]: 對椰！謝謝提醒，已調整 host 為 "host": "35.72.177.254:3000"

**Comment by nauish** [comment]
我沒有問題~

**Pull request merged**